main.go

```diff
@@ -277,76 +274,79 @@ func last_seen_output(last_seen []int64, now time.Time) (output string) {
+	log.Printf("200: /info for %v", house_id)
 }
 
-func target_mac_handler(res http.ResponseWriter, req *http.Request) {
-	query := req.URL.Query()
-	api_key := query.Get("api_key")
-	house_id_string := query.Get("id")
-	house_id, _ := strconv.ParseInt(house_id_string, 0, 64)
-	if validate_key(api_key, int(house_id)) {
-		target_mac := Common.target_mac[house_id]
-		target_mac_binary := mactobinary(target_mac)
-		res.Header().Set("Content-Type", "application/octet-stream")
-		res.Write(target_mac_binary)
-		log.Printf("200: target_mac: %v ", target_mac)
-	} else {
-		http.Error(res, "403 Forbidden : you can't access this resource.", 403)
-		log.Printf("403: /target_mac from %v, using api key %v", house_id, api_key)
+func handle_target_mac(res http.ResponseWriter, req *http.Request) {
+	house_id := validate_key(res, req)
+	if house_id < 0 {
+		return
 	}
+	target_mac := Common.target_mac[house_id]
+	target_mac_binary := mactobinary(target_mac)
+	res.Header().Set("Content-Type", "application/octet-stream")
+	res.Write(target_mac_binary)
+	log.Printf("200: target_mac: %v ", target_mac)
 }
 
-func turn_on(res http.ResponseWriter, req *http.Request) {
-	query := req.URL.Query()
-	api_key := query.Get("api_key")
-	house_id_string := query.Get("id")
-	house_id, _ := strconv.ParseInt(house_id_string, 0, 64)
-	if validate_key(api_key, int(house_id)) {
-		Common.Set(int(house_id), true)
-		fmt.Fprintf(res, "Turned on %v", house_id)
-		log.Printf("200: turn_on: %v", house_id)
-	} else {
-		http.Error(res, "403 Forbidden : you can't access this resource.", 403)
-		log.Printf("403: /on from %v, using api key %v", house_id, api_key)
+func handle_turn_on(res http.ResponseWriter, req *http.Request) {
+	house_id := validate_key(res, req)
+	if house_id < 0 {
+		return
 	}
+	Common.Set(int(house_id), true)
+	fmt.Fprintf(res, "Turned on %v", house_id)
+	log.Printf("200: turn_on: %v", house_id)
 }
 
-func turn_off(res http.ResponseWriter, req *http.Request) {
-	query := req.URL.Query()
-	api_key := query.Get("api_key")
-	house_id_string := query.Get("id")
-	house_id, _ := strconv.ParseInt(house_id_string, 0, 64)
-	if validate_key(api_key, int(house_id)) {
-		Common.Set(int(house_id), false)
-		fmt.Fprintf(res, "Turned off %v", house_id)
-		log.Printf("200: turn_off: %v", house_id)
-	} else {
-		http.Error(res, "403 Forbidden : you can't access this resource.", 403)
-		log.Printf("403: /off from %v, using api key %v", house_id, api_key)
+func handle_turn_off(res http.ResponseWriter, req *http.Request) {
+	house_id := validate_key(res, req)
+	if house_id < 0 {
+		return
+	}
+	Common.Set(int(house_id), false)
+	fmt.Fprintf(res, "Turned off %v", house_id)
+	log.Printf("200: turn_off: %v", house_id)
+}
+
+func handle_log(res http.ResponseWriter, req *http.Request) {
+	house_id := validate_key(res, req)
+	if house_id < 0 {
+		return
+	}
+	out, err := exec.Command("./etherhoused-log", strconv.FormatInt(house_id, 10)).Output()
```

Comment:
I don't mind the shelling out to the script, but the fact that the actual contents of the script is the original "make logs", it confuses me, and I think would confuse others.

I would prefer the server run something *not* in git, and git-ignore it maybe. In real life /log would never work for real as is.